File: 003/003.php
<?php

class Dictionary {
    private $dict;

    public function __construct() {
        $this->dict = [];
    }

    public function insert($key, $value) {
        $this->dict[$key] = $value;
    }

    public function print() {
        echo "{";
        
        $counter = 1;
        
        foreach ($this->dict as $key => $value) {
            echo "$key: $value";

            if ($counter != count($this->dict)) {
                echo ", ";
            }

            $counter++;
        }

        echo "}" . PHP_EOL;
    }
}

function genDictionary($max_key) {
    $dictionary = new Dictionary();

    for ($k = 1; $k <= $max_key; $k++) {
        $dictionary->insert($k, $k * $k);
    }

    return $dictionary;
}

// *************************************************** //

if (!function_exists("readline")) {
    function readline($message) {
        $fh = fopen("php://stdin", "r");
        echo $message;

        $userInput = trim(fgets($fh));
        fclose($fh);

        return $userInput;
    }
}

$max_key = (int)readline("Enter a number: ");
$dictionary = genDictionary($max_key);
$dictionary->print();

?>

File: 003/003.cpp
#include <iostream>
#include <map>

using namespace std;

map<int, int> genDictionary(int max_key) {
    map<int, int> dictionary;

    for (int i = 1; i <= max_key; i++) {
        dictionary[i] = i * i;
    }

    return dictionary;
}

void printDictionary(map<int, int> dictionary) {
    map<int, int>::iterator iter;

    cout << "{";

    int counter = 1;

    for (iter = dictionary.begin(); iter != dictionary.end(); iter++) {
        cout << (*iter).first << ": " << (*iter).second;
        if (counter != dictionary.size()) {
            cout << ", ";
        }
        counter++;
    }

    cout << "}" << endl;
}

int main() {
    string user_input;

    cout << "Enter a number: ";
    cin >> user_input;

    int max_key = stoi(user_input);

    map<int, int> dictionary = genDictionary(max_key);
    printDictionary(dictionary);
    
    return 0;
}